world.js

```diff
@@ -303,7 +303,14 @@ const _bindHitTracker = app => {
   };
   world.appManager.addEventListener('frame', frame);
   const die = () => {
-    world.appManager.removeTrackedApp(app.instanceId);
+    if (app.npcPlayer?.avatar?.fsm?.states?.ragdoll) {
+      app.npcPlayer.avatar.fsms.send('ragdoll');
+      setTimeout(() => {
```

Comment:
Why timeout?

Reply:
If no timeout, the npc will stop updating physics and removed from the scene immediately.
So has no time for ragdoll to play death animation.

It's best to watch the deactive state of all ragdoll bones instead of setTimeout, I'll try.
But using hardcoded time may worth as the first version.

Reply:
Makes sense!